modules/kubernetes-addons/argocd/locals.tf

```diff
@@ -60,4 +58,9 @@ locals {
     account     = data.aws_caller_identity.current.account_id
     clusterName = var.eks_cluster_id
   }
+
+  admin_password = var.admin_password_secret_name == "" ? "" : data.aws_secretsmanager_secret_version.admin_password_secret_version[0].secret_string
+  default_helm_values = [templatefile("${path.module}/values.yaml", {
+    admin_password : local.admin_password == "" ? "" : bcrypt(local.admin_password, 10)
+  })]
```

Comment:
We can change this block to leverage override sensitive values. This will ensure the default config is not overwritten by user supplied config. 

```hcl
  override_set_sensitive_values = var.admin_password_secret_name != "" ? [
    {
      name  = "configs.secret.argocdServerAdminPassword"
      value = bcrypt(data.aws_secretsmanager_secret_version.admin_password_secret_version[0].secret_string, 10)
    }
    ]: []
    
    #  Now this variable(`local.override_set_sensitive_values`) can be referred in Helm Chart. See comment below 
```

`main.tf`

```hcl
  dynamic "set_sensitive" {
    iterator = each_item
    for_each = distinct(concat(local.override_set_sensitive_values, local.helm_config["set_sensitive"]))

    content {
      name  = each_item.value.name
      value = each_item.value.value
    }
  }
```

Reply:
Done